### PR TITLE
#41 - Refactor Repomd.performUpdate for clarity

### PR DESCRIPTION
**Pull request by victornoel**
This is for #41, I believe more can be done so I added another todo.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#58](https://github.com/artipie/rpm-adapter/issues/58) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/a9ab9d00-b9b2-43ac-a4d5-71ab5d592a83); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by Vatavuk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/rpm-adapter/issues/58#issuecomment-606200542))

@Vatavuk The user @vatavuk[/z](https://www.0crat.com/u/vatavuk) resigned from [#58](https://github.com/artipie/rpm-adapter/issues/58), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/rpm-adapter/issues/58#issuecomment-606200542))

@Vatavuk Job refused in 6 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
This pull request [#58](https://github.com/artipie/rpm-adapter/issues/58) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/1be3a9d0-3ac7-449c-ab30-82772c2ef315); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/Repomd.java [hunk 1]
```diff
@@ -139,6 +139,12 @@ private Completable loadRepomd(final Path file) {
      * @param act Action
      * @param repomd The temp
      * @return Completion or error signal
+     * @todo #41:30min Continue refactoring Repomd.performUpdate method
```
Comment (paulodamaso) [inline src/main/java/com/artipie/rpm/Repomd.java]: @victornoel You could improve the architecture right now instead just suggesting it, for example, instead just putting this puzzle saying "we might move some parts of this into other classes" you could put puzzles in the parts of the code that you think that could be in another classes and ask the user to implement this new design

**Comment by victornoel** [comment]
@paulodamaso thank you, I've tried something, tell me what you think.

**Comment by victornoel** [comment]
@g4s8 I've merged master into this branch to be on top of master, but now the build is failing, what am I missing? Should I rebase instead?

**Comment by g4s8** [comment]
@victornoel please check build logs: https://github.com/artipie/rpm-adapter/pull/58/checks?check_run_id=549729114

some checkstyle rules failed

**Comment by victornoel** [comment]
@g4s8 indeed, sorry, I saw a warning at the beginning and got stuck on that :)

@paulodamaso it should be ok now

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@paulodamaso please make sure you will find at least 3 issues during next CR, thanks

**Comment by paulodamaso** [comment]
@sereshqua Ok I will, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**